src/main/scala/subsystem/CrossingWrapper.scala

```diff
@@ -158,36 +160,39 @@ trait HasCrossingMethods extends LazyModule with LazyScope
   def crossIntRationalIn (alreadyRegistered: Boolean = false)(implicit p: Parameters): IntNode = crossIntRationalInOut(false)(alreadyRegistered)
   def crossIntRationalOut(alreadyRegistered: Boolean = false)(implicit p: Parameters): IntNode = crossIntRationalInOut(true )(alreadyRegistered)
 
-  def crossIntIn(arg: SubsystemClockCrossing, alreadyRegistered: Boolean)(implicit p: Parameters): IntNode = arg match {
+  def crossIntIn(alreadyRegistered: Boolean)(implicit p: Parameters): IntNode = arg match {
     case x: SynchronousCrossing  => crossIntSyncIn(alreadyRegistered)
     case x: AsynchronousCrossing => crossIntAsyncIn(x.sync, alreadyRegistered)
     case x: RationalCrossing     => crossIntRationalIn(alreadyRegistered)
   }
 
-  def crossIntOut(arg: SubsystemClockCrossing, alreadyRegistered: Boolean)(implicit p: Parameters): IntNode = arg match {
+  def crossIntOut(alreadyRegistered: Boolean)(implicit p: Parameters): IntNode = arg match {
     case x: SynchronousCrossing  => crossIntSyncOut(alreadyRegistered)
     case x: AsynchronousCrossing => crossIntAsyncOut(x.sync, alreadyRegistered)
     case x: RationalCrossing     => crossIntRationalOut(alreadyRegistered)
   }
 
-  def crossIntIn (arg: SubsystemClockCrossing)(implicit p: Parameters): IntNode = crossIntIn (arg, false)
-  def crossIntOut(arg: SubsystemClockCrossing)(implicit p: Parameters): IntNode = crossIntOut(arg, false)
+  def crossIntIn (implicit p: Parameters): IntNode = crossIntIn (false)
+  def crossIntOut(implicit p: Parameters): IntNode = crossIntOut(false)
 }
 
-trait HasCrossing extends HasCrossingMethods
+
+trait HasCrossing extends LazyScope
 {
   this: LazyModule =>
-  val crossing: SubsystemClockCrossing
 
-  def crossTLIn   (implicit p: Parameters): TLNode  = crossTLIn   (crossing)
-  def crossTLOut  (implicit p: Parameters): TLNode  = crossTLOut  (crossing)
-  def crossAXI4In (implicit p: Parameters): AXI4Node= crossAXI4In (crossing)
-  def crossAXI4Out(implicit p: Parameters): AXI4Node= crossAXI4Out(crossing)
-  def crossIntIn  (implicit p: Parameters): IntNode = crossIntIn  (crossing)
-  def crossIntOut (implicit p: Parameters): IntNode = crossIntOut (crossing)
+  def crossing: SubsystemClockCrossing
+  protected lazy val xing = new CrossingHelper(this, crossing)
```

Comment:
This should not be lazy. The class has no effects, so there is no reason to be lazy. Making it lazy will make it called 'xing_lzy' in signal names. We can skip the _lzy I think.

Reply:
It's for backwards compatibility with things following the `Island` pattern that declare a hard-coded `val crossing` in their constructor body rather than taking it as a constructor parameter.